<?xml version="1.0"?>
<launch>
	<arg name="teleop" default="joy" />
	<arg name="filter" default="on"/>
	<arg name="viz" default="on"/>

	<include file="$(find segway_sim)/launch/rviz.launch" if="$(eval viz == 'on')"/>

	<group ns="segway_sim">

		<node name="integrator" pkg="segway_sim" type="integrator_node" output="screen">
			<param name="dt" value="0.001" />
			<param name="umax" value="20.0" />
			<param name="input_delay_ms" value="10.0"/>
			<rosparam param="IC">[0.5, 5.5, 3.1415, 0.0, 0.0, 0.0, 0.0]</rosparam>
		</node>


		<node name="momdp_python" pkg="segway_sim" type="momdp_python_obst.py" output="screen">
			<!-- 5x5 -->
			<param name="testToTry" value="-1"/>
			<!-- 7x7 -->
			<!-- <param name="testToTry" value="-2"/> -->
			<!-- 10x10 -->
			<!-- <param name="testToTry" value="-3"/> -->
		</node>


		<node name="sensor" pkg="segway_sim" type="sensor_node" output="screen">
			<param name="offset_angle" value="0.138324423615"/>
		</node>

		<node name="ekf" pkg="segway_sim" type="ekf_node" output="screen">
			<param name="offset_angle" value="0.138324423615"/>
		</node>

		<node name="low_level" pkg="segway_sim" type="low_level_node" output="screen">
			<param name="offset_angle" value="0.138324423615"/>
			<param name="low_level_input_delay" value="0.0"/>

			<param name="xmax"        type="double" value="0.04"/>
			<param name="ymax"        type="double" value="0.04"/>
			<param name="thetamax"    type="double" value="0.1"/>
			<param name="vmax"        type="double" value="0.3"/>
			<param name="thetaDotmax" type="double" value="0.3"/>
			<param name="psimax"      type="double" value="0.1"/>
			<param name="psiDotmax"   type="double" value="0.3"/>

			<param name="low_level_active" value="0.0"/>
		</node>

		<node name="mpc" pkg="segway_sim" type="mpc_node" output="screen">
			<param name="offset_angle" value="0.138324423615"/>
			<param name="mpc_input_delay" value="-1.0"/>
			<param name="x_start" value="0.5"/>

			<param name="y_start" value="5.5"/>
			<param name="theta_start" value="3.14"/>
			<!-- IMPORTANT: the effect of the delay in the prediction horizon is a bit approximated when delay =! dt_mpc (if needed can be handles more carefully) -->
			<param name="e_x"        type="double" value="0.04"/>
			<param name="e_y"        type="double" value="0.04"/>
			<param name="e_theta"    type="double" value="0.1"/>
			<param name="e_v"        type="double" value="0.3"/>
			<param name="e_thetaDot" type="double" value="0.3"/>
			<param name="e_psi"      type="double" value="0.1"/>
			<param name="e_psiDot"   type="double" value="0.3"/>

			<param name="enlarge"    type="double" value="0.0"/>		
		</node>

		<node pkg="rosbag" type="record" name="record" output="screen" args="record -a -o /home/ugo/rosbag/"/> 
	
	</group>
</launch>


<!-- rosservice call /segway_sim/integrator/ui "cmd: 1" -->